src/main/java/com/minecolonies/colony/Colony.java

```diff
@@ -702,15 +708,47 @@ private void spawnCitizen(CitizenData data)
             entity.setPosition(spawnPoint.getX(), spawnPoint.getY(), spawnPoint.getZ());
             world.spawnEntityInWorld(entity);
 
+            checkAchievements();
+
             markCitizensDirty();
         }
     }
 
     /**
+     * Checks if the achievements are valid.
+     */
+    private void checkAchievements()
+    {
+        // the colonies size
+        final int size = this.citizens.size();
+
+
+        ArrayList<Consumer<EntityPlayer>> consumers = new ArrayList<>();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'consumers' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)